src/components/form/Form.tsx

```diff
@@ -139,3 +155,62 @@ function useSetExpandedWidth() {
     setExpandedWidth(defaultExpandedWidth);
   }, [currentPageId, expandedPagesFromLayout, expandedWidthFromSettings, setExpandedWidth]);
 }
+
+const emptyArray = [];
+interface ErrorProcessingProps {
+  setFormState: React.Dispatch<React.SetStateAction<FormState>>;
+}
+
+/**
+ * Instead of re-rendering the entire Form component when any of this changes, we just report the
+ * state to the parent component.
+ */
+function ErrorProcessing({ setFormState }: ErrorProcessingProps) {
+  const currentPageId = useCurrentView();
+  const topLevelNodeIds = useNodesMemoSelector(
+    (nodes) =>
+      nodes
+        .findLayout(currentPageId)
+        ?.children()
+        .map((n) => n.item.id) || emptyArray,
+  );
```

Comment:
Is there a reason the empty array is stored in a const and only referenced? Is it because we dont want to create a new empty array every time?

Reply:
Yup, correct. Because `[] !== []`, every time it renders the empty array would be a different one. When looking it over once more, I'm not sure it does much here, but I usually do that as a safer default when in doubt.